### PR TITLE
[Layer] Add requireLabel() function

### PR DESCRIPTION
**Pull request by zhoonit**
- [Layer] Add requireLabel() function

```
Add a simple function `Layer::requireLayer()`.
This function is used to detect if a certain layer requires a layer.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

**Review by kparichay — APPROVED** [review]
LGTM

File: nntrainer/layers/layer_internal.h [hunk 1]
```diff
@@ -126,6 +126,19 @@ class Layer {
...
+  /**
+   * @brief  check if this layer requires label to be passed
+   * @note   if requireLabel() == true means, for now, that it is endpoint of a
+   * graph(numOutlayers == 0). label will be fed to the gradient of hidden if
+   * requireLabel is true
+   * @todo   If we get to have a use case for requireLabel(true) but in the
+   * middle of a graph, change the semantics
+   *
+   * @return true requires a label when training
+   * @return false does not require a label
```
Comment (myungjoo) [inline nntrainer/layers/layer_internal.h:138]: If you describe w/ a return value and its description, @retval is more appropriate. 

In general, use @return if you describe without any structure, use @retval if you pair up "return value" with its description.

Comment (zhoonit) [reply nntrainer/layers/layer_internal.h:138]: Thank you for the explanation :). There are number of this issue in other places, so I'll make a separate PR for this.

File: nntrainer/layers/layer_internal.h [hunk 2]
```diff
@@ -126,6 +126,19 @@ class Layer {
...
+   */
+  virtual bool requireLabel() { return false; }
```
Comment (kparichay) [inline nntrainer/layers/layer_internal.h:140]:
```suggestion
  virtual bool requireLabel() const { return false; }
```

File: nntrainer/layers/layer_internal.h (hunk 1 above)
Comment (zhoonit) [reply nntrainer/layers/layer_internal.h:138]: #1264 deals the issue :)

File: nntrainer/layers/layer_internal.h (hunk 2 above)
Comment (zhoonit) [reply nntrainer/layers/layer_internal.h]: Done

**Review by myungjoo — APPROVED** [review]
LGTM

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**